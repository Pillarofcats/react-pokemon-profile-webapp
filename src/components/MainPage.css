.App {
  background-color: rgb(228, 74, 74);
  text-align: center;
  min-height: 100vh;
  color: gold;
  /* min-width: 200px; */
}

.Search-container {
  min-height: 150px;
  /* min-width: 200px; */
  /* padding-bottom: 10px; */
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  font-size: 20px;
  background-color: rgb(228, 74, 74);
  position: fixed;
  top: 0;
  width: 100%;
}