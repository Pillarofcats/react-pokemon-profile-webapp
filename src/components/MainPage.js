//Dependencies
import React, {Component} from 'react';
//Components
import Header from "../components/Header.js";
import Scroll from "../components/Scroll.js";
import Profiles from "../components/Profiles.js";
import Search from "../components/Search.js";
import ErrorBoundary from "../components/ErrorBoundary.js";
//Styles
import './MainPage.css';

class MainPage extends Component {

  componentDidMount() {
    // //Get data from a server and initialize state with it
    this.props.onRequestPokemon();
  };

  filteredPokemon = () => {
    //Filter through each cat profile that matches our current searching state and return those cat profiles
    return this.props.pokemon_list.filter(pokemon => {

      return pokemon.name.toLowerCase().includes(this.props.searching.toLowerCase());
    });
  };

  render() {
    //Destructure state variables
    const {onSearchChange, isPending} = this.props;
    
      // else if(pokemon.id.toLowerCase().includes(searching.toLowerCase())) {
      //   return true;
      // }

    return (
      <div className="App">
            <div className = "Search-container">
              <Header />
              <Search 
                searchingOnChange = {onSearchChange}
              />
            </div>
            {/* <Scroll> */}
              { isPending ? <h1>Loading..</h1> :
                // <div className = 'Profiles-container'>
                  /* <ErrorBoundary> */
                    <Profiles pokemon = {this.filteredPokemon()}/>
                  /* </ErrorBoundary> */
                // </div>
              }
            {/* </Scroll> */}
      </div>
        );
  }
};
//Connect App to React-Redux store using {connect}
export default MainPage;
