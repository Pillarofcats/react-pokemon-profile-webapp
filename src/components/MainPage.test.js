import React from 'react';
import {shallow} from 'enzyme';
import MainPage from "./MainPage.js";


//Test 1
let wrapper;

//Runs function before tests are run and initializes props for <MainPage {...mockProps} />
beforeEach(() => {
  const mockProps = {
    onRequestPokemon: jest.fn(),
    pokemon_list: [],
    searching: '',
    isPending: false
  }

  //Set Enzyme shallow function to wrapper to be tested with Jest expect assertion tests
  wrapper = shallow(<MainPage {...mockProps}/>)
});

//Test render the wrapper function with 'shallow(<MainPage {...mockProps}/>'
it("Renders MainPage without Crashing", ()=> {
  expect(wrapper).toMatchSnapshot();
});


//Test 2
let wrapper2;

it("Filter Pokemon Correctly", () => {

  const mockProps2 = {
    onRequestPokemon: jest.fn(),
    pokemon_list: [
      {
        name: 'bulbasaur',
        id: 0
      }
    ],
    searching: 'bulb',
    isPending: false
  }

  //Set Enzyme shallow function to wrapper to be tested with Jest expect assertion tests
  wrapper2 = shallow(<MainPage {...mockProps2}/>)

  expect(wrapper2.instance().filteredPokemon()).toEqual([
    {
      name: 'bulbasaur',
      id: 0
    }]);
});


//Test 3
let wrapper3;

it("Filter Pokemon Correctly Again", () => {

  const mockProps3 = {
    onRequestPokemon: jest.fn(),
    pokemon_list: [
      {
        name: 'bulbasaur',
        id: 0
      }
    ],
    searching: 'bulb',
    isPending: false
  }

  //Set Enzyme shallow function to wrapper to be tested with Jest expect assertion tests
  const pokemon = [{
        name: 'bulbasaur',
        id: 0
      }];
      
  wrapper3 = shallow(<MainPage {...mockProps3}/>)

  expect(wrapper3.instance().filteredPokemon()).toEqual(pokemon);
});