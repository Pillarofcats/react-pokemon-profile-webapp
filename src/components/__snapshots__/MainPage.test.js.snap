// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders MainPage without Crashing 1`] = `ShallowWrapper {}`;
